From Yinghai.Lu@Sun.COM Sat Feb  2 22:17:42 2008
From: Yinghai Lu <Yinghai.Lu@Sun.COM>
Date: Sat, 02 Feb 2008 22:24:47 -0800
Subject: PCI: pcie AER: don't check _OSC when acpi is disabled
To: Andrew Morton <akpm@linux-foundation.org>
Cc: linux-kernel@vger.kernel.org, Len Brown <lenb@kernel.org>, greg@kroah.com
Message-ID: <200802022224.47400.yinghai.lu@sun.com>
Content-Disposition: inline


[PATCH] pcie AER: don't check _OSC when acpi is disabled

when acpi=off or pci=noacpi, get warning

AER service couldn't init device 0000:00:0a.0:pcie01 - no _OSC support
AER service couldn't init device 0000:00:0e.0:pcie01 - no _OSC support
AER service couldn't init device 0000:00:0f.0:pcie01 - no _OSC support
AER service couldn't init device 0000:80:0b.0:pcie01 - no _OSC support
AER service couldn't init device 0000:80:0e.0:pcie01 - no _OSC support
AER service couldn't init device 0000:80:0f.0:pcie01 - no _OSC support

so don't check _OSC in aer_osc_setup

Signed-off-by: Yinghai Lu <yinghai.lu@sun.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pcie/aer/aerdrv_acpi.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/pci/pcie/aer/aerdrv_acpi.c
+++ b/drivers/pci/pcie/aer/aerdrv_acpi.c
@@ -33,6 +33,9 @@ int aer_osc_setup(struct pcie_device *pc
 	struct pci_dev *pdev = pciedev->port;
 	acpi_handle handle = 0;
 
+	if (acpi_pci_disabled)
+		return -1;
+
 	/* Find root host bridge */
 	while (pdev->bus && pdev->bus->self)
 		pdev = pdev->bus->self;
